apps/condo/domains/miniapp/schema/B2CApp.js

```diff
@@ -0,0 +1,148 @@
+            type: Relationship,
+            ref: 'B2CAppBuild',
+            isRequired: false,
+            kmigratorOptions: { null: true, on_delete: 'models.PROTECT' },
+            hooks: {
+                validateInput: async ({ resolvedData, fieldPath, addFieldValidationError, operation, existingItem }) => {
+                    const appId = operation === 'create' ? resolvedData.id : existingItem.id
+                    if (resolvedData.hasOwnProperty(fieldPath) && resolvedData[fieldPath] !== null) {
+                        const validBuild = await getByCondition('B2CAppBuild', {
+                            deletedAt: null,
+                            app: { id: appId },
+                            id: resolvedData[fieldPath],
+                        })
+                        if (!validBuild) {
+                            return addFieldValidationError(RESTRICT_BUILD_SELECT_ERROR)
+                        }
+                    }
+                },
+            },
+        },
+        builds: {
+            schemaDoc: 'List of available app builds. Removing item from this list automatically soft-deletes it',
+            type: Relationship,
+            ref: 'B2CAppBuild.app',
+            many: true,
+            hooks: {
+                validateInput: async ({ resolvedData, fieldPath, addFieldValidationError, operation, existingItem }) => {
+                    const appId = operation === 'create' ? resolvedData.id : existingItem.id
+                    const buildIds = resolvedData[fieldPath]
+                    const restrictedBuilds = await find('B2CAppBuild', {
+                        id_in: buildIds,
+                        app: { id_not: appId },
+                        deletedAt: null,
+                    })
+                    if (restrictedBuilds.length) {
+                        return addFieldValidationError(RESTRICT_BUILD_APP_CHANGE_ERROR)
+                    }
+                },
+            },
+        },
+        properties: {
+            schemaDoc: 'List of all B2C app properties where B2C app can be launched',
+            type: Relationship,
+            ref: 'B2CAppProperty.app',
+            many: true,
+            hooks: {
+                validateInput: async ({ resolvedData, fieldPath, addFieldValidationError, operation, existingItem }) => {
+                    const appId = operation === 'create' ? resolvedData.id : existingItem.id
+                    const propertyIds = resolvedData[fieldPath]
+                    const restrictedProperties = await find('B2CAppProperty', {
+                        id_in: propertyIds,
+                        app: { id_not: appId },
+                        deletedAt: null,
+                    })
+                    if (restrictedProperties.length) {
+                        return addFieldValidationError(RESTRICT_PROPERTY_APP_CHANGE_ERROR)
+                    }
+                },
+            },
+        },
+        accessRights: {
```

Comment:
It's better to give access to this field only for admins. It's access

Reply:
removed

Reply:
You can't fetch any accessRIght if you're not admin / support....And specifying additional access here will not remove it from GQL types as well as not improve query performance....So, I'll left it as it was